tests/e2e/e2e_test.go

```diff
@@ -184,6 +184,72 @@ var _ = Describe("Backstage Operator E2E", func() {
 			})
 		}
 	})
+
+	Context("Operator upgrade with existing instances", Label("upgrade"), func() {
```

Comment:
If the goal is to validate default config when upgrade to current version from old ones (which is a valid goal) can we consider making some tiny "framework" like (for example) iterate testing the older scripts with predictable names from a directory like
testdata/
   upgrade/
       backstage-operator-0.1.x.yaml
       backstage-operator-0.2.x.yaml
       
And do the test for each script in this directory (I would also move this upgrade test to dedicated  file and make an instruction)

This way we can easily manage the scenario by adding/removing (when the version is no longer supported) these yaml scripts.

WDYT?

Reply:
Yeah, that makes sense. I thought about a similar approach when adding this `backstage-operator-x.y.z.yaml` file and including such "upgrade" tests as part of the nightly Workflow. But due to the current severity of this bug, I think this can be a later improvement, which can even be assigned to one of our newcomers to get started.
I'll create a separate issue for this.